components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +98,31 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    /**
+     *
+     * @param fedIdpSessionId
+     * @return A map containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    public Map<String, String> getSessionDetails(String fedIdpSessionId) throws SessionManagementServerException {
```

Comment:
Do we need a separate method here?
What happens if we use the getSession() method with the federatedSessionID?

Reply:
We can't use the getSession() to retrieve the federated authentication session info.